ledger/ledger-utxo/src/main/java/net/corda/v5/ledger/utxo/query/json/ContractStateVaultJsonFactory.java

```diff
@@ -0,0 +1,67 @@
+package net.corda.v5.ledger.utxo.query.json;
+
+import net.corda.v5.application.marshalling.JsonMarshallingService;
+import net.corda.v5.ledger.utxo.ContractState;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * The interface that needs to be implemented in order to create a JSON representation of a state type. Implementing
```

Comment:
Should mention that the JSON stored here is to be used by the vault named queries. We have to make sure that the link between this factory and the named queries is 100% clear.